### ISSUE TITLE
`--config` is ignored in directories that contains `.swiftlint.yml`

### ISSUE THREAD

**Issue by norio-nomura** [realm/SwiftLint#550]
e.g.

``` sh
% cd SwiftLint

# use `force_unwrapping` enabled configuration
% cat ~/.swiftlint-test.yml
opt_in_rules:
  - force_unwrapping

# create a file expecting generate a `force_unwrapping` violation
% echo 'let url = NSURL(string: query)!'>550.swift

# using `--use-stdin` generate violation
% cat 550.swift|swiftlint lint --config ~/.swiftlint-test.yml --use-stdin
Loading configuration from '/Users/norio/.swiftlint-test.yml'
<nopath>:1:31: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.

# using `--path` does not generate violation
% swiftlint lint --config ~/.swiftlint-test.yml --path 550.swift
Loading configuration from '/Users/norio/.swiftlint-test.yml'
Linting Swift files at path 550.swift
Linting '550.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.

# enable `force_unwrapping` in `.swiftlint.yml`
% echo '  - force_unwrapping'>>.swiftlint.yml

# using `--path` generate violation by using `.swiftlint.yml`
% swiftlint lint --config ~/.swiftlint-test.yml --path 550.swift
Loading configuration from '/Users/norio/.swiftlint-test.yml'
Linting Swift files at path 550.swift
Linting '550.swift' (1/1)
/Users/norio/Documents/workspace/github/SwiftLint/550.swift:1:31: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by scottrhoyt** [comment]
Hmm, this sounds like it might be related to the enabling `use_nested_configs` by default. #527 

At first blush it seems like if nested configurations is the default, then it will be hard to solve this, particularly for using `--config non_root_of_linting_directories/.swiftlint.yml` or if the expectation is that the `--config` will override all `.swiftlint.yml`s found while linting. I guess if only the latter is a consideration, then we could disabled nested configurations when a config is specified via CLI.

**Comment by jpsim** [comment]
Still an issue.

**Comment by marcelofabri** [comment]
This seems to be fixed on 0.20.1.

```
$ cat 550.swift|swiftlint lint --config ~/.swiftlint-test.yml --use-stdin --no-cache
Loading configuration from '/Users/marcelofabri/.swiftlint-test.yml'
<nopath>:1:31: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.

$ swiftlint lint --config ~/.swiftlint-test.yml --path 550.swift --no-cache
Loading configuration from '/Users/marcelofabri/.swiftlint-test.yml'
Linting Swift files at path 550.swift
Linting '550.swift' (1/1)
/Users/marcelofabri/550/550.swift:1:31: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.

$ swiftlint lint --config ~/.swiftlint-test.yml --path 550.swift --no-cache
Loading configuration from '/Users/marcelofabri/.swiftlint-test.yml'
Linting Swift files at path 550.swift
Linting '550.swift' (1/1)
/Users/marcelofabri/550/550.swift:1:31: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.
```
